### ISSUE TITLE
Support Authenticated (and/or Conditional) requests to prevent GitHub API rate limiting

### ISSUE THREAD

**Issue by jsundqvist** [craigburke/client-dependencies-gradle#44]
Hi!

I'm running `clientInstall` with the following configuration

```
clientDependencies {
    bower {
        'polymer'('1.4.0') { source '*' }
        'app-route'('0.9.1', url: 'PolymerElements/app-route') { source '*' }
        'paper-input'('1.0.0', url: 'PolymerElements/paper-input') { source '*' }
    }
}
```

and get the following response after successfully downloading 15 dependencies

```
{
  "message": "API rate limit exceeded for 10.101.110.11. (But here's the good news: Authenticated requests get a higher rate limit. Check out the documentation for more details.)",
  "documentation_url": "https://developer.github.com/v3/#rate-limiting"
}
```

Is there a way to supply user credentials to prevent rate limiting?

> For requests using Basic Authentication or OAuth, you can make up to 5,000 requests per hour. For unauthenticated requests, the rate limit allows you to make up to 60 requests per hour. Unauthenticated requests are associated with your IP address, and not the user making requests. 

Follow-up question which might (not) be a separate issue:
Does the plugin support conditional requests?

> Most responses return an ETag header. Many responses also return a Last-Modified header. You can use the values of these headers to make subsequent requests to those resources using the If-None-Match and If-Modified-Since headers, respectively. If the resource has not changed, the server will return a 304 Not Modified. **Also note: making a conditional request and receiving a 304 response does not count against your Rate Limit, so we encourage you to use it whenever possible.**

Thanks again.

/Johannes

**Comment by craigburke** [comment]
Good idea. I definitely should do this.

**Comment by craigburke** [comment]
@jsundqvist I'll hopefully have time to do this in the next couple days but I'd probably recommend using npm as your source instead of bower anyway (that's what I use). You won't run into this rate limit issue there.

**Comment by craigburke** [comment]
@jsundqvist in `1.2.0` I'm changing the default behavior so that the client side dependency aren't being constantly re-installed (calling `./gradlew clean` used to wipe out the client dependency forcing them to be reinstalled). 

How I've been using this, is that I'm committing the client dependencies in my project and running `./gradlew clientRefresh` only when I'm updating my dependencies. 

I'll comment here when I've released `1.2.0`. I can revisit this if it remains a problem with this new workflow but I'm closing this for now.

**Comment by craigburke** [comment]
@jsundqvist I just released `1.2.0` let me know if that helps.

**Comment by sbglasius** [comment]
I have a `githubUsername`/`githubPassword` solution almost ready. I will do a PR shortly, but still need to get some tests in place, if I can find a way to do it.
